src/lib/http/http_server.pl

```diff
@@ -122,9 +122,11 @@
     default(StatusCode0, 200, StatusCode),
     maplist(map_header_kv_2, ResponseHeaders, ResponseHeaders0),
     '$http_answer'(ResponseHandle, StatusCode, ResponseHeaders, ResponseStream),
-    call_cleanup(
-	format(ResponseStream, "~s", [ResponseText]),
-	close(ResponseStream)
+    chars_utf8bytes(ResponseText, ResponseBytes),
+    catch(
+	call_cleanup(maplist(put_byte(ResponseStream), ResponseBytes),close(ResponseStream)),
```

Comment:
This is very expensive, since it writes every byte individually. Is there a way to use `format/2` for this? It seems that should work.

Note that `chars_utf8bytes/2` is also very expensive, and forfeits the compact internal string representation!

Reply:
This was actually what I needed to change in order to not generate corrupted responses (as you might remember from IRC). However, I'm going to try to change the type of the stream as you told me

Reply:
Thank you a lot! There could very well be another, unrelated, mistake hidden in Scryer's stream or string handling that is now only overshadowed by this workaround. `format/3` or `phrase_to_stream/2` should work, and if not, then we should correct this! I think a good test case tries, over and over, to send and receive messages, for many hours.